src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -144,6 +146,63 @@ static void create_static_function_call(
   }
 }
 
+/// If \p argument_for_this contains a dereference then create a temporary
+/// variable for it and use that instead
+/// \param [in,out] argument_for_this: The first argument of the function call
+/// \param symbol_table: The symbol table to add the new symbol to
+/// \param vcall_source_loc: The source location of the function call, which is
+///   used for new instructions that are added
+/// \param [out] new_code_for_this_argument: New instructions are added here
+static void process_this_argument(
+  exprt &argument_for_this,
+  symbol_table_baset &symbol_table,
+  const source_locationt &vcall_source_loc,
+  goto_programt &new_code_for_this_argument)
+{
+  const auto dereference_it = std::find_if(
+    argument_for_this.depth_cbegin(),
+    argument_for_this.depth_cend(),
+    [](const exprt &expr) { return can_cast_expr<dereference_exprt>(expr); });
```

Comment:
You could use `const bool argument_for_this_contains_a_dereference = has_subexpr(argument_for_this, ID_dereference);` If you don't like the way `has_subexpr` is implemented then you might still use `has_subexpr` here, but also replace the implementation of `has_subexpr` by your code.